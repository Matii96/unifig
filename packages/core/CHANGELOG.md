# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.2.0 (2022-10-12)

### Bug Fixes

- **core:** added missing class-validator dependency ([6052855](https://github.com/Matii96/unifig/commit/60528552db7396b967907cd262cecb3612f7293a))

### Features

- **core:** improved manager interface ([cef08c7](https://github.com/Matii96/unifig/commit/cef08c77b380c5c063969821829adc309705f927))
- **core:** package setup ([cb2a066](https://github.com/Matii96/unifig/commit/cb2a066f4d40dfa31779cb187067f7134204ac3b))
- **core:** throwing exceptions upon encountering problems ([86b48d7](https://github.com/Matii96/unifig/commit/86b48d7166ba79a75755ade0a791271277d7c521))
- **env:** package setup ([5221aef](https://github.com/Matii96/unifig/commit/5221aeff7ac7b071619be5f6c9e537bb74cdacb8))

### BREAKING CHANGES

- **core:** changed getters names to more descriptive equivalents
