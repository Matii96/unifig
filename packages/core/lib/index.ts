export * from './adapters';
export * from './manager';
export * from './properties-mapping';
export * from './validator';
