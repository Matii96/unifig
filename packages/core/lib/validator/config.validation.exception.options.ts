import { ValidationError } from 'class-validator';
import { Type } from '../utils/type.interface';

export interface ConfigValidationExceptionOptions {
  template: Type<any>;
  errors: ValidationError[];
}
