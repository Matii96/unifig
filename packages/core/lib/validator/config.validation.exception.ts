export class ConfigValidationException<TConfig extends object> extends Error {
  constructor(config: TConfig, msg: string) {
    super(`Validation failed for ${config.constructor.name}:${msg}`);
    this.name = ConfigValidationException.name;
  }
}
