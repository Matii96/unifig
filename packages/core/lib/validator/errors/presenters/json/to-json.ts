import { ConfigValidationError } from '../../validation.error';
import { ToJSONOptions } from './to-json.options';

export const toJSON = (
  validationException: ConfigValidationError,
  options: ToJSONOptions = {},
): string =>
  JSON.stringify(
    validationException.errors.map((templateError) => ({
      template: templateError.template.name,
      errors: templateError.errors,
    })),
    null,
    options.space,
  );
