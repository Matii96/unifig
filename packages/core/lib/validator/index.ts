export * from './decorators';
export * from './errors';
