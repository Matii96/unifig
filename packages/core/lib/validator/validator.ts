import { ConfigValidationError } from './errors/validation.error';

export interface Validator {
  validate(configs: object[]): ConfigValidationError | undefined;
}
