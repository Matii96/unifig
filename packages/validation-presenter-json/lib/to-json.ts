import { ConfigValidationException } from '@unifig/core';
import { ToJSONOptions } from './to-json.options';

export const toJSON = (validationException: ConfigValidationException, options: ToJSONOptions = {}): string =>
  JSON.stringify(
    validationException.errors.map((templateError) => ({
      template: templateError.template.name,
      errors: templateError.errors,
    })),
    null,
    options.space
  );
