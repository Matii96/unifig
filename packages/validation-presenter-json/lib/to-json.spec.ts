import { ConfigValidationException } from '@unifig/core';
import { mockFailed1dValidation } from './validator.mocks';
import { toJSON } from './to-json';

describe('toJSON', () => {
  it('should format json from 1d error', () => {
    expect(toJSON(new ConfigValidationException([mockFailed1dValidation()]))).toMatchSnapshot();
  });
});
