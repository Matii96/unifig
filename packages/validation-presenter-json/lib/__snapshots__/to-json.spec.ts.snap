// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toJSON should format json from 1d error 1`] = `"[{"template":"StorageOptions","errors":[{"property":"port","source":"PORT","currentValue":"not-a-port","failedConstraints":[{"details":"port must be an integer number","name":"isInt"}]}]}]"`;
